lightning/src/ln/channelmanager.rs

```diff
@@ -2409,38 +2465,58 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
-					ChannelUpdateStatus::EnabledStaged if !chan.is_live() => chan.set_channel_update_status(ChannelUpdateStatus::Disabled),
-					ChannelUpdateStatus::DisabledStaged if !chan.is_live() => {
-						if let Ok(update) = self.get_channel_update_for_broadcast(&chan) {
-							channel_state.pending_msg_events.push(events::MessageSendEvent::BroadcastChannelUpdate {
-								msg: update
-							});
-						}
-						should_persist = NotifyOption::DoPersist;
-						chan.set_channel_update_status(ChannelUpdateStatus::Disabled);
-					},
-					ChannelUpdateStatus::EnabledStaged if chan.is_live() => {
-						if let Ok(update) = self.get_channel_update_for_broadcast(&chan) {
-							channel_state.pending_msg_events.push(events::MessageSendEvent::BroadcastChannelUpdate {
-								msg: update
-							});
-						}
-						should_persist = NotifyOption::DoPersist;
-						chan.set_channel_update_status(ChannelUpdateStatus::Enabled);
-					},
-					_ => {},
-				}
+			let new_feerate = self.fee_estimator.get_est_sat_per_1000_weight(ConfirmationTarget::Normal);
+
+			let mut handle_errors = Vec::new();
+			{
+				let mut channel_state_lock = self.channel_state.lock().unwrap();
+				let channel_state = &mut *channel_state_lock;
+				let pending_msg_events = &mut channel_state.pending_msg_events;
+				let short_to_id = &mut channel_state.short_to_id;
+				channel_state.by_id.retain(|chan_id, chan| {
+					match chan.channel_update_status() {
+						ChannelUpdateStatus::Enabled if !chan.is_live() => chan.set_channel_update_status(ChannelUpdateStatus::DisabledStaged),
+						ChannelUpdateStatus::Disabled if chan.is_live() => chan.set_channel_update_status(ChannelUpdateStatus::EnabledStaged),
+						ChannelUpdateStatus::DisabledStaged if chan.is_live() => chan.set_channel_update_status(ChannelUpdateStatus::Enabled),
+						ChannelUpdateStatus::EnabledStaged if !chan.is_live() => chan.set_channel_update_status(ChannelUpdateStatus::Disabled),
+						ChannelUpdateStatus::DisabledStaged if !chan.is_live() => {
+							if let Ok(update) = self.get_channel_update_for_broadcast(&chan) {
+								pending_msg_events.push(events::MessageSendEvent::BroadcastChannelUpdate {
+									msg: update
+								});
+							}
+							should_persist = NotifyOption::DoPersist;
+							chan.set_channel_update_status(ChannelUpdateStatus::Disabled);
+						},
+						ChannelUpdateStatus::EnabledStaged if chan.is_live() => {
+							if let Ok(update) = self.get_channel_update_for_broadcast(&chan) {
+								pending_msg_events.push(events::MessageSendEvent::BroadcastChannelUpdate {
+									msg: update
+								});
+							}
+							should_persist = NotifyOption::DoPersist;
+							chan.set_channel_update_status(ChannelUpdateStatus::Enabled);
+						},
+						_ => {},
+					}
+
+					let counterparty_node_id = chan.get_counterparty_node_id();
+					let (retain_channel, chan_needs_persist, err) = self.update_channel_fee(short_to_id, pending_msg_events, chan_id, chan, new_feerate);
```

Comment:
Maybe update `timer_tick_occured` doc in consequences ? At least it turns a `chain::Watch::update_chan` call frequency from a rare event to one function of mempool fluctuations?